### PR TITLE
Feature/external adapter

### PR DESCRIPTION
**Pull request by zackwoo808**

### CODE REVIEW

File: StravaExternalAdapter/index.js [hunk 1]
```diff
@@ -1,73 +1,167 @@
...
+        })
+    })
+    .catch(500, error => {
+      // console.log(error, 'error on DB access')
+      cb(500, Requester.errored(jobRunID, error));
+    });
+}
+// #endregion
+
+// #region wrapper functions
```
Comment (zackwoo808) [inline StravaExternalAdapter/index.js:131]: haven't exposed these yet

File: StravaExternalAdapter/app.js [hunk 2]
```diff
@@ -1,18 +1,40 @@
...
+      const { distance } = result.data.result || {};
+
+      if (isNaN(distance) && !distance) {
+        return res.status(404).json(result)
+      }
+
+      res.status(200).json({
+        distance,
+        id,
+        timestamp: Date.now() // TODO
```
Comment (zackwoo808) [inline StravaExternalAdapter/app.js:34]: not sure what type of timestamp we want?

File: StravaExternalAdapter/app.js [hunk 3]
```diff
@@ -1,18 +1,49 @@
...
 
 const express = require('express')
 const bodyParser = require('body-parser')
 const app = express()
 const port = process.env.EA_PORT || 8080
 
-app.use(bodyParser.json())
+app.use(bodyParser.json());
 
-app.post('/', (req, res) => {
```
Comment (tannervelten) [inline StravaExternalAdapter/app.js:10]: We need this

File: StravaExternalAdapter/app.js [hunk 4]
```diff
@@ -1,18 +1,49 @@
...
-  console.log('POST Data: ', req.body)
-  createRequest(req.body, (status, result) => {
-    console.log('Result: ', result)
-    res.status(status).json(result)
-  })
-})
+app.post('/get-athlete-activity', (req, res) => {
```
Comment (tannervelten) [inline StravaExternalAdapter/app.js:15]: Don't need this endpoint, business logic can go in https://github.com/thearthurchen/TogetherWeRun/blob/main/StravaExternalAdapter/strava.js#L4

File: StravaExternalAdapter/app.js [hunk 5]
```diff
@@ -1,18 +1,49 @@
...
+      if (isNaN(distance) && !distance) {
+        return res.status(404).json(result)
+      }
+
+      res.status(200).json(distance);
+    });
+  });
+});
+
+app.post('/create-new-user', (req, res) => {
```
Comment (tannervelten) [inline StravaExternalAdapter/app.js:37]: Good

File: StravaExternalAdapter/index.js [hunk 6]
```diff
@@ -1,73 +1,221 @@
-const { Requester, Validator } = require('@chainlink/external-adapter')
```
Comment (tannervelten) [inline StravaExternalAdapter/index.js:1]: This file handles the req/res for EA -- you shouldn't need to make changes here

**Review by tannervelten — CHANGES_REQUESTED** [review]
Adding this in case you wanna take a look, but we can review later. Think I dropped the ball a bit on explaining where to add your code

**Pull request closed without merging**